src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -3196,66 +3196,97 @@ instruct vcvtBtoX(vReg dst, vReg src) %{
     BasicType bt = Matcher::vector_element_basic_type(this);
     if (is_floating_point_type(bt)) {
       __ integer_extend_v(as_VectorRegister($dst$$reg), bt == T_FLOAT ? T_INT : T_LONG,
-                          Matcher::vector_length(this), as_VectorRegister($src$$reg), T_BYTE);
+                          Matcher::vector_length(this), as_VectorRegister($src$$reg), T_BYTE,
+                          true /* is_signed */);
       __ csrwi(CSR_FRM, C2_MacroAssembler::rne);
       __ vfcvt_f_x_v(as_VectorRegister($dst$$reg), as_VectorRegister($dst$$reg));
     } else {
       __ integer_extend_v(as_VectorRegister($dst$$reg), bt,
-                          Matcher::vector_length(this), as_VectorRegister($src$$reg), T_BYTE);
+                          Matcher::vector_length(this), as_VectorRegister($src$$reg), T_BYTE,
+                          true /* is_signed */);
     }
   %}
   ins_pipe(pipe_slow);
 %}
 
-// VectorCastS2X
+// VectorUCastB2X
 
-instruct vcvtStoB(vReg dst, vReg src) %{
-  predicate(Matcher::vector_element_basic_type(n) == T_BYTE);
-  match(Set dst (VectorCastS2X src));
-  format %{ "vcvtStoB $dst, $src" %}
+instruct vcvtUBtoX(vReg dst, vReg src) %{
+  match(Set dst (VectorUCastB2X src));
+  effect(TEMP_DEF dst);
+  format %{ "vcvtUBtoX $dst, $src" %}
   ins_encode %{
-    __ integer_narrow_v(as_VectorRegister($dst$$reg), T_BYTE, Matcher::vector_length(this),
-                        as_VectorRegister($src$$reg), T_SHORT);
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    if (is_floating_point_type(bt)) {
```

Comment:
Could `bt` (the vector element basic type) be floating point type for `VectorUCastB2X` node? I see our aarch64  counterpart has this assertion: `assert(bt == T_SHORT || bt == T_INT || bt == T_LONG, "must be");` [1]. Same question for `VectorUCastS2X` and `VectorUCastI2X` nodes.

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/aarch64_vector.ad#L3752

Reply:
Yeh, seems it's not, and vector api does not have this operations either.
Fixed, thanks for catching.